Inventory.md

```diff
@@ -0,0 +1,173 @@
+- [Other, fuzzier difficulties of integration](#other-fuzzier-differences-of-integration)
+- [Summary Byron](#Summary-Byron)
+- [Extra: A glimpse into Shelley](#extra-A-glimpse-into-shelley)
+
+## Inventory
+
+### Protocol Parameters & Constants
+
+These are the protocol parameters and constants the wallet is currently using:
+
+| Parameter | Description | Exact equivalent in Haskell Byron Node |
+| ----- |:-------------| --- |
+| genesisBlockHash | Hash of the very first block | ✔ |
+| genesisBlockDate | Start time of the chain. | ✔ |
+| feePolicy | Policy regarding transaction fee. | ❌ |
+| slotLength | Length, in seconds, of a slot. | ✔ |
+| epochLength | Number of slots in a single epoch. | ✔ |
+| txMaxSize | Maximum size of a transaction (soft or hard limit). | ✔ |
+| epochStability | Length of the suffix of the chain considered unstable | ✔ |
+| activeSlotCoefficient | In Genesis/Praos, corresponds to the % of active slots (i.e. slots for which someone can be elected as leader). | ❌ |
+
+Genesis and slotting information should trivially be the same. 
+
+The fee policies work differently. 
+
+Jörmungandr charges `constant + coeff * (inputs + outputs) + optional_cert_specific_fee`
+
+The Haskell Byron node charges `constant + coeff * tx_size_in_bytes`
+
+some links:
+
+- [Wallet BlockchainParameters](https://github.com/input-output-hk/cardano-wallet/blob/468e0fb14ed3b50ab3d16ba127f24242a41ed126/lib/core/src/Cardano/Wallet/Primitive/Types.hs#L1084-L1102)
+- [cardano-ledger ProtocolParameters](https://github.com/input-output-hk/cardano-ledger/blob/0ad59ce46b3da8af6f6af97064e7c77cc7c2ac8b/cardano-ledger/src/Cardano/Chain/Update/ProtocolParameters.hs#L32-L62)
+- [wallet use of Jörmungandr fee policy](https://github.com/input-output-hk/cardano-wallet/blob/468e0fb14ed3b50ab3d16ba127f24242a41ed126/lib/jormungandr/src/Cardano/Wallet/Jormungandr/Transaction.hs#L105)
+- [doc of Jörmungandr fee policy: format.md](https://github.com/input-output-hk/chain-libs/blob/master/chain-impl-mockchain/doc/format.md)
+- [cardano-ledger TxFeePolicy](https://github.com/input-output-hk/cardano-ledger/blob/0ad59ce46b3da8af6f6af97064e7c77cc7c2ac8b/cardano-ledger/src/Cardano/Chain/Common/TxFeePolicy.hs#L41-L59)
+
+
+### Protocol Parameter Updates
+
+Jörmungandr does not fully support protocol parameter updates. If protocol parameters can change, the wallet needs to be aware.
+
+- Jörmungandr: static (updates not implemented)
+- Haskell Byron: static (updates possible, but not likely)
+- Haskell Shelley: dynamic (updates possible, wallet should handle this)
+
+### Blocks
+
+The Jörmungandr wallet uses the `BlockHeader = (slotId, blockHeight, hash, prevHash)` record to identify points on the chain.
+
+The Byron Haskell node only *needs* `Point ByronBlock = Origin | At (slotNo, hash)` to start giving us blocks. The wallet can retrieve `blockHeight` and `prevHash` from block headers. But those two fields are not needed to initiate the chain-sync protocol. Small ❌. 
```

Comment:
I don't really see a problem here to be honest. If our type can do more, then it can certainly do less. We aren't looking for the minimal implementation possible; integrating with multiple backends necessarily create some overhead.

Reply:
> We aren't looking for the minimal implementation possible; integrating with multiple backends necessarily create some overhead.

We need to convert `BlockHeader` to `Origin | At (slotNo, headerHash)`. What special values should correspond to the `Origin` case? Could this be an implementation detail of the DB? Does this now need to be known inside `NetworkLayer`? If we are converting `SlotId -> SlotNo` we now depend on dynamic protocol parameters (though I guess we are likely to change that).

I think we can get by. Still, I *am* — *slightly* — worried that not being abstract enough now could lead to unnecessary complexity later on.

My despair from adding jörmungandr support vaguely comes to mind:

> To add jormungadnr support we had to
> - come up with a tip that works with all targets (BlockHeader)
> - Modify the *core* Wallet s t state & walletlayer etc to support that tip
> - change NetworkLayer to support that tip (edited)

[src](https://input-output-rnd.slack.com/archives/DBHFW31K7/p1560445363080300)

Reply:
- we know if a block is the genesis block (i.e. origin) by comparing its hash with the known genesis hash
- SlotId -> SlotNo is a transformation that needs to happen regardless. So, not really a problem. Our `BlockHeader` type is currently a superset of this `Origin | At (slotNo headerHash)` data-type.